client.go

```diff
@@ -0,0 +1,77 @@
+package compute
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+type Client struct {
+	config     *Config
+	httpClient *http.Client
+}
+
+func NewClient(config *Config, httpClient *http.Client) (*Client, error) {
+	if config == nil {
+		return nil, errors.New("failed to NewClient: config cannot be nil")
+	}
+
+	if httpClient == nil {
+		return nil, errors.New("failed to NewClient: httpClient cannot be nil")
+	}
+
+	client := &Client{
+		config:     config,
+		httpClient: httpClient,
+	}
+
+	return client, nil
+}
+
+func NewLocalClient() (*Client, error) {
+	return NewClient(LocalConfig(), http.DefaultClient)
+}
+
+func (c *Client) NewRunnable(environment, customerID, namespace, fnName string) *Runnable {
```

Comment:
I'm unsure about the `NewRunnable` naming here because it sounds like the client is performing some action to create a runnable somewhere, which is not true. Maybe just `Runnable` ? Or does it even need to be a method on `Client` now that it doesn't hold a reference to that client?

Reply:
`Runnable` probably shouldn’t even have this type of constructor then. Would it be reasonable to expect all runnables to be created as a result to an API call? Either UserFunctions or a builder endpoint